GliaWidgets/Component/Button/Header/HeaderButton.swift

```diff
@@ -13,19 +13,27 @@ class HeaderButton: UIButton {
 
     private let style: HeaderButtonStyle
     private let kSize = CGSize(width: 30, height: 30)
+    private var isDefineLayoutNeeded = true
 
     init(with style: HeaderButtonStyle) {
         self.style = style
         super.init(frame: .zero)
         setup()
-        layout()
     }
 
     @available(*, unavailable)
     required init?(coder _: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
+    override func updateConstraints() {
+        defer { super.updateConstraints() }
```

Comment:
Interesting, aren't we supposed to call this first? Or in this particular case, we should call it last?

Reply:
yep, because we expect that `defineLayout` function will create all necessary constraints inside of view and after they have been created we will call `super.updateConstraints` to update layout. This flow will respect possible specific with constraints in current view.